### PR TITLE
Add call-back functions for integrated BC

### LINKED ISSUE #17206: IntegratedBC is missing some callback functions
## Reason
Not like `Kernel`, who has calls to `precalculateResidual`, `IntegratedBC` is missing some of call-back functions. This make derived class unable to custom its evaluation unless overriding `computeResidual` function, which we would like to avoid.

## Design
Add the call-back functions for `IntegratedBC`.

## Impact
New feature and should not have impact on current capabilities.

### PR DESCRIPTION
**Pull request by YaqiWang**
Close #17206.

### CODE REVIEW

File: framework/include/bcs/IntegratedBC.h [hunk 1]
```diff
@@ -67,6 +67,22 @@ class IntegratedBC : public IntegratedBCBase, public MooseVariableInterface<Real
     return computeQpOffDiagJacobian(jvar);
   }
 
+  /**
+   * Put necessary evaluations depending on qp but independent on test functions here
```
Comment (loganharbour) [inline framework/include/bcs/IntegratedBC.h:71]:
```suggestion
   * Insertion point for evaluations that depend on qp but are independent of the test functions.
```

Comment (loganharbour) [reply framework/include/bcs/IntegratedBC.h:71]: Trying to make some of this consistent. Would you mind changing the doc string for the methods that follow in a similar manner?

If you have the time - I'd also appreciate copying this to DGKernel. I think I missed this last time.

**Review by loganharbour — CHANGES_REQUESTED** [review]
Good other than the docstrings

File: framework/include/bcs/IntegratedBC.h (hunk 1 above)
Comment (YaqiWang) [reply framework/include/bcs/IntegratedBC.h]: ok

**Review by loganharbour — APPROVED** [review]
Thanks @YaqiWang for being willing and expeditious with the changes.

**Comment by YaqiWang** [comment]
My pleasure. Thanks for the review.

**Pull request merged**